routemaster/state_machine/api.py

```diff
@@ -219,3 +237,66 @@ def delete_label(app: App, label: LabelRef) -> None:
             old_state=current_state.name,
             new_state=None,
         ))
+
+
+LabelStateProcessor = Callable[[App, State, StateMachine, LabelRef, Any], bool]
+
+
+def process_cron(
+    process: LabelStateProcessor,
+    get_labels: Callable[[StateMachine, State, Any], Iterable[str]],
+    app: App,
+    state_machine: StateMachine,
+    state: State,
+    should_terminate: IsExitingCheck,
```

Comment:
I would advocate not threading `should_terminate` through and instead doing this by short-circuit logic in the iterable returned by `get_labels`. It simplifies quite a bit of this code and avoids mocking in the tests.

Reply:
I finally managed this in 91dd677. I'm not sure it's better overall though – I think `process_job` might need wrapping up in a class that gets attributes set through the process perhaps?